server/bin/index-pbench

```diff
@@ -2095,112 +2486,299 @@ class PbenchTarBall(object):
         # compressed with a different compression mechanism from xz?
         self.dirname = dirname[:dirname.rfind('.tar.xz')]
-        # ... but let's make sure.
-        sampled_prefix = self.tb.getmembers()[0].name.split(os.path.sep)[0]
-        if sampled_prefix != self.dirname:
-            # All members of the tarball should have self.dirname as its
-            # prefix.
+        # ... but let's make sure ...
+        #
+        # ... while we are at it, we verify we have a metadata.log file in the
+        # tar ball before we start extracting.
+        metadata_log_path = "%s/metadata.log" % (self.dirname)
+        metadata_log_found = False
+        self.members = self.tb.getmembers()
+        for m in self.members:
+            if m.name == metadata_log_path:
+                metadata_log_found = True
+            sampled_prefix = m.name.split(os.path.sep)[0]
+            if sampled_prefix != self.dirname:
+                # All members of the tarball should have self.dirname as its
+                # prefix.
+                raise UnsupportedTarballFormat(
+                    "{} - directory prefix should be \"{}\", but is"
+                    " \"{}\" instead, for tarball member \"{}\"".format(
+                        self.tbname,
+                        self.dirname,
+                        sampled_prefix,
+                        m.name))
+        if not metadata_log_found:
             raise UnsupportedTarballFormat(
-                "{} - tarball member dir prefix should be \"{}\", but is \"{}\" instead.".format(
-                    self.tbname,
-                    self.dirname,
-                    sampled_prefix))
-
-        # Verify we have a metadata.log file in the tar ball before we
-        # start extracting.
-        member_name = "%s/metadata.log" % (self.dirname)
-        try:
-            self.tb.getmember(member_name)
-        except KeyError:
-            raise UnsupportedTarballFormat(
-                "{} - tarball is missing \"{}\".".format(self.tbname, member_name))
+                "{} - tarball is missing \"{}\".".format(
+                    self.tbname, metadata_log_path))
 
         # We are expected to clean up the extracted tar ball before we exit.
-        self.tb.extractall(path=tmpdir)
-        self.extracted_root = tmpdir
+        self.extracted_root = self.idxctx.options.tmpdir
+        self.tb.extractall(path=self.extracted_root)
         if not os.path.isdir(os.path.join(self.extracted_root, self.dirname)):
             self.delete_extracted()
             raise UnsupportedTarballFormat(
                 "{} - extracted tarball but can't get to the directory \"{}\".".format(
                     self.tbname,
                     os.path.join(self.extracted_root, self.dirname)))
+        # Open the MD5 file of the tarball and read the MD5 sum from it.
+        md5sum = open("%s.md5" % (self.tbname)).read().split()[0]
```

Comment:
Yes, which means it will raise an exception killing the PbenchTarBall() object construction at line 3157, resulting an exception that will be caught at line 3218, reported as an unexpected exception.  The `pbench-dispatch` could is supposed to have validated that we have a `.md5` file along side this tar ball, so it seems appropriate to leave it as a course-grain exception handling path.